gitlabform/lists/projects.py

```diff
@@ -112,6 +99,36 @@ def _get_projects(self, target: str, groups: Groups) -> Projects:
                 OmissionReason.ARCHIVED,
                 archived_projects_from_configuration_not_from_groups,
             )
+        else:
+            # in all other cases, we also need to look for projects in the config
+            # that are being transferred to a different namespace
+            for project in projects_from_configuration_not_from_groups:
+                # if the target is a group, and the project is not in the group
+                if (
+                    len(groups.get_effective()) == 1
+                    and target != project.rsplit("/", 1)[0]
+                ):
```

Comment:
What happens when a group contains a subgroup? For example:

```py
>>> project='group1/subgroup1-1/project1'
>>> print(project.rsplit("/", 1)[0])
group1/subgroup1-1
```

When we do `gitlabform group1`, does this mean projects in `subgroup1-1` will not be processed? I feel like it should be processed. What do you think @long-wan-ep .

This does seem like related to issue #613 . Don't want to bloat this PR with multiple issues, but if you agree with what I said above and if it's a simple couple lines of changes, should we just do it? On the other hand, not sure if this is considered as a breaking change. If yes, then maybe we go with what you have.

@gdubicki - would you mind chiming in here if available?

Reply:
Yes, I believe only projects in the group, not subgroup, are processed. This is the default GitLab API behaviour.

Reply:
> This is the default GitLab API behaviour.

While that is true, role of gitlabform is automation. It processes individual groups and projects individually using corresponding API endpoint anyways, but iterates through list or target projects to process them all or the ones defined. So, I feel the subgroups should also be included. Otherwise it's an unintuitive behaviour. If anyone doesn't want a subgroup to be processed, they can use the `skip_groups` key in the config to control what subgroup to be excluded. Or, maybe an additional CLI flag to indicate all subgroup processing should be excluded.

This feels like a potentially breaking change. So, best not to handle it in this PR.